test/Microsoft.ML.Functional.Tests/Evaluation.cs

```diff
@@ -0,0 +1,279 @@
+using Microsoft.ML.TestFramework;
+using Microsoft.ML.Trainers;
+using Microsoft.ML.Trainers.FastTree;
+using Microsoft.ML.Trainers.KMeans;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Functional.Tests
+{
+    public class Evaluation : BaseTestClass
+    {
+        public Evaluation(ITestOutputHelper output): base(output)
+        {
+        }
+
+        /// <summary>
+        /// Train and Evaluate: Anomaly Detection.
+        /// </summary>
+        [Fact]
+        public void TrainAndEvaluateAnomalyDetection()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+
+            var trainData = mlContext.Data.ReadFromTextFile<MnistOneClass>(GetDataPath(TestDatasets.mnistOneClass.trainFilename),
+                hasHeader: TestDatasets.mnistOneClass.fileHasHeader,
+                separatorChar: TestDatasets.mnistOneClass.fileSeparator);
+            var testData = mlContext.Data.ReadFromTextFile<MnistOneClass>(GetDataPath(TestDatasets.mnistOneClass.testFilename),
+                hasHeader: TestDatasets.mnistOneClass.fileHasHeader,
+                separatorChar: TestDatasets.mnistOneClass.fileSeparator);
+
+            // Create a training pipeline.
+            var pipeline = mlContext.AnomalyDetection.Trainers.RandomizedPca();
+
+            // Train the model.
+            var model = pipeline.Fit(trainData);
+
+            // Evaluate the model.
+            //  TODO #2464: Using the train dataset will cause NaN metrics to be returned.
+            var scoredTest = model.Transform(testData);
+            var metrics = mlContext.AnomalyDetection.Evaluate(scoredTest);
+
+            // Check that the metrics returned are valid.
+            Common.CheckMetrics(metrics);
+        }
+
+        /// <summary>
+        /// Train and Evaluate: Binary Classification.
+        /// </summary>
+        [Fact]
+        public void TrainAndEvaluateBinaryClassification()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+
+            var data = mlContext.Data.ReadFromTextFile<TweetSentiment>(GetDataPath(TestDatasets.Sentiment.trainFilename),
+                hasHeader: TestDatasets.Sentiment.fileHasHeader,
+                separatorChar: TestDatasets.Sentiment.fileSeparator);
+
+            // Create a training pipeline.
+            var pipeline = mlContext.Transforms.Text.FeaturizeText("Features", "SentimentText")
+                .AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.BinaryClassification.Trainers.LogisticRegression(
+                    new LogisticRegression.Options { NumThreads = 1 }));
+
+            // Train the model.
+            var model = pipeline.Fit(data);
+
+            // Evaluate the model.
+            var scoredData = model.Transform(data);
+            var metrics = mlContext.BinaryClassification.Evaluate(scoredData);
+
+            // Check that the metrics returned are valid.
+            Common.CheckMetrics(metrics);
```

Comment:
>CheckMetrics [](start = 19, length = 12)

So the reason why I mentioned `CalibratedBinaryClassificationMetrics` is not only to add them to Common.cs, but also to have test on them.
Maybe have two pipelines in this test. One with calibrated, one without.

Reply:
Gotcha ;)

---
In reply to: [259509132](https://github.com/dotnet/machinelearning/pull/2646#discussion_r259509132) [](ancestors = 259509132)